squidpy/pl/_spatial_utils.py

```diff
@@ -0,0 +1,897 @@
+    return sorted(set.intersection(*map(set, dic.values())))
+
+
+def _get_list(var: Any, _type: Any, ref_len: int | None = None) -> List[Any] | Any:
+    if isinstance(var, _type):
+        if ref_len is None:
+            return [var]
+        return [var] * ref_len
+    elif isinstance(var, list):
+        if (ref_len is not None) and (ref_len != len(var)):
+            raise ValueError(f"Var len: {len(var)} is not equal to ref len: {ref_len}. Please Check.")
+        return var
+    else:
+        raise ValueError(f"Can't make list from var: `{var}`")
+
+
+def _set_color_source_vec(
+    adata: AnnData,
+    value_to_plot: str | None,
+    use_raw: bool | None = None,
+    alt_var: str | None = None,
+    layer: str | None = None,
+    groups: _SeqStr = None,
+    palette: Palette_t = None,
+    na_color: str | Tuple[float, ...] | None = None,
+) -> Tuple[NDArrayA | pd.Series, NDArrayA, bool]:
+
+    if alt_var is not None and value_to_plot not in adata.obs.columns and value_to_plot not in adata.var_names:
+        value_to_plot = adata.var_names[adata.var[alt_var] == value_to_plot][0]
+    if use_raw and value_to_plot not in adata.obs.columns:
+        color_source_vector = adata.raw.obs_vector(value_to_plot)
+    else:
+        color_source_vector = adata.obs_vector(value_to_plot, layer=layer)
+    if groups is not None and is_categorical_dtype(color_source_vector):
+        color_source_vector = color_source_vector.replace(color_source_vector.categories.difference(groups), np.nan)
+
+    to_hex = partial(colors.to_hex, keep_alpha=True)
+    if value_to_plot is None:
+        return None, np.full(to_hex(na_color), adata.n_obs), False
+    if not is_categorical_dtype(color_source_vector):
+        return None, color_source_vector, False
+    else:
+        clusters = color_source_vector.categories
+        color_map = _get_palette(adata, cluster_key=value_to_plot, categories=clusters, palette=palette)
+        color_vector = color_source_vector.rename_categories(color_map)
+        # Set color to 'missing color' for all missing values
+        if color_vector.isna().any():
+            color_vector = color_vector.add_categories([to_hex(na_color)])
+            color_vector = color_vector.fillna(to_hex(na_color))
+        return color_source_vector, color_vector, True
+
+
+def _shaped_scatter(
+    x: NDArrayA,
+    y: NDArrayA,
+    s: float,
+    c: NDArrayA,
+    shape: _AvailShapes | None = ScatterShape.CIRCLE.s,  # type: ignore[assignment]
+    norm: _Normalize = None,
+    **kwargs: Any,
+) -> PatchCollection:
+    """
+    Get shapes for scatterplot.
+
+    Adapted from here: https://gist.github.com/syrte/592a062c562cd2a98a83 .
+    This code is under [The BSD 3-Clause License](http://opensource.org/licenses/BSD-3-Clause)
+    """
+    shape = ScatterShape(shape)  # type: ignore[assignment]
+    if TYPE_CHECKING:
+        assert isinstance(shape, ScatterShape)
+    shape = shape.s
+
+    if shape == ScatterShape.CIRCLE.s:
```

Comment:
Change to `shape == ScatterShape.CIRCLE`

Reply:
inheritance as suggested above doesn't work